### ISSUE TITLE
Support for 64-bit integers

### ISSUE THREAD

**Issue by Jylpah** [Ironholds/primes#17]
Hello, 

Is there a way to get the package to work with 64bit integers (`bit64` package)? When trying to generate larger primes, I get an error message that hints towards 32bit integer support only.

```R
> generate_primes(min=1000000, max=10000000000)
integer(0)
Warning message:
In generate_primes(min = 1e+06, max = 1e+10) :
  NAs introduced by coercion to integer range
>
``` 

Or 

```R
> prime_factors(5e12+1)
[[1]]
[1] NA

Warning message:
In prime_factors(5e+12 + 1) : NAs introduced by coercion to integer range
```

**Comment by pegeler** [comment]
Hi @Jylpah,

Thanks for using our package! I'm sorry that it appears its capabilities don't align with your use case. You are correct that all functionality is limited to signed 32-bit integers, which is a consequence of R's internal implementation of integer vectors.

The [`gmp`](https://cran.r-project.org/package=gmp) package supports integers of arbitrary size and has some builtin functions for working with primes (`isprime()` and `nextprime()`). This would be my first suggestion for working with large prime numbers. I'm not aware of any reverse imports of `gmp` that provide functions for generating prime factors, though (however, I have not looked very hard).

I believe that switching to a 3rd party data type would be outside the scope of this package. The `primes` package is meant to provide a wide array of tools for working with prime numbers using relatively well-tuned Rcpp code. The limited domain was a known trade-off for speed and ease of use. To that end, I think we'll need to stick with native data types here. However, let me know if there is a use-case that `gmp` and its derivatives do not cover and perhaps I could build a package around that, if it seems tractable :smiley:.

**Comment by Jylpah** [comment]
Thank you for your reply. I managed to do what I was intending to do with the `primes` package. In the end I did not need 64-bit primes, but it was enough to use 64-bit integers for the result of multiplying a "large" 32-bit integer with "large" 32-bit primes. _With mathematicians one needs to be careful what to call "large" :-)_ 

And yes, the `primes` package is very easy to use and comes with many handy functions. Thank you for that.
